From 1925845dc10330e4b48fec68333fac6ef2b7bf5c Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Wed, 1 Jun 2022 08:56:08 +0900
Subject: [PATCH] test: import logind test from debian/ubuntu test suite

[dtardon: Picked just the scaffolding, not the tests themselves.]

(cherry picked from commit 9c94ab0f6ff22da4278a6e9a93ddc480607c55ac)

Related: #1866955
---
 test/TEST-35-LOGIN/Makefile     |  1 +
 test/TEST-35-LOGIN/test.sh      | 55 +++++++++++++++++++++++++++++++++
 test/TEST-35-LOGIN/testsuite.sh |  9 ++++++
 3 files changed, 65 insertions(+)
 create mode 120000 test/TEST-35-LOGIN/Makefile
 create mode 100755 test/TEST-35-LOGIN/test.sh
 create mode 100755 test/TEST-35-LOGIN/testsuite.sh

diff --git a/test/TEST-35-LOGIN/Makefile b/test/TEST-35-LOGIN/Makefile
new file mode 120000
index 0000000000..e9f93b1104
--- /dev/null
+++ b/test/TEST-35-LOGIN/Makefile
@@ -0,0 +1 @@
+../TEST-01-BASIC/Makefile
\ No newline at end of file
diff --git a/test/TEST-35-LOGIN/test.sh b/test/TEST-35-LOGIN/test.sh
new file mode 100755
index 0000000000..32410c8149
--- /dev/null
+++ b/test/TEST-35-LOGIN/test.sh
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -e
+
+TEST_DESCRIPTION="LOGIN"
+
+# shellcheck source=test/test-functions
+. "${TEST_BASE_DIR:?}/test-functions"
+
+test_setup() {
+    create_empty_image
+    mkdir -p "$TESTDIR"/root
+    mount "${LOOPDEV}p1" "$TESTDIR"/root
+
+    (
+        LOG_LEVEL=5
+        eval $(udevadm info --export --query=env --name="${LOOPDEV}p2")
+
+        setup_basic_environment
+
+        inst_binary pkill
+        inst_binary useradd
+        inst_binary userdel
+
+        # mask some services that we do not want to run in these tests
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-hwdb-update.service
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-journal-catalog-update.service
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-networkd.service
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-networkd.socket
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-resolved.service
+        ln -fs /dev/null "$initdir"/etc/systemd/system/systemd-machined.service
+
+        # setup the testsuite service
+        cat >"$initdir"/etc/systemd/system/testsuite.service <<EOF
+[Unit]
+Description=Testsuite service
+
+[Service]
+ExecStart=/bin/bash -x /testsuite.sh
+Type=oneshot
+StandardOutput=tty
+StandardError=tty
+NotifyAccess=all
+EOF
+        cp testsuite.sh "$initdir"/
+
+        setup_testsuite
+    ) || return 1
+    setup_nspawn_root
+
+    ddebug "umount $TESTDIR/root"
+    umount "$TESTDIR"/root
+}
+
+do_test "$@"
diff --git a/test/TEST-35-LOGIN/testsuite.sh b/test/TEST-35-LOGIN/testsuite.sh
new file mode 100755
index 0000000000..9855b4bc80
--- /dev/null
+++ b/test/TEST-35-LOGIN/testsuite.sh
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+set -o pipefail
+
+: >/failed
+
+touch /testok
+rm /failed
